src/valid/function.rs

```diff
@@ -843,7 +837,14 @@ impl super::Validator {
 
         #[cfg(feature = "validate")]
         for (index, argument) in fun.arguments.iter().enumerate() {
-            match module.types[argument.ty].inner.pointer_class() {
+            let ty = module.types.get_handle(argument.ty).ok_or(
+                FunctionError::InvalidArgumentType {
```

Comment:
This isn't the right error for this situation. `InvalidArgumentType` is supposed to be for types that Naga forbids passing into functions, not wild type handles.

I think we should just have one error, `BadArenaHandle { arena: &'static str, holder: &'static str, index: usize }`, and not bother providing any more detail about it, since the only people who care about these errors are front end developers. I guess we can't put `BadArenaHandle` in `ValidationError`, since the context provided by wrapping `FunctionError`, `TypeError`, etc. in a `ValidationError` is valuable. Maybe it's good enough to just add identical `BadArenaHandle` variants to each of the more specific error enums.